projects/angular/src/forms/common/if-control-state/if-control-state.service.ts

```diff
@@ -34,11 +35,26 @@ export class IfControlStateService implements OnDestroy {
           // Subscribe to the status change events, only after touched
           // and emit the control
           this.subscriptions.push(
-            this.control.statusChanges.subscribe(() => {
+            this.control.statusChanges?.subscribe(() => {
               this.triggerStatusChange();
             })
           );
         }
+      }),
+
+      this.ngControlService.additionalControlChanges.subscribe((controls: NgControl[]) => {
+        if (controls) {
+          this.additionalControls = controls;
+          // Subscribe to the status change events, only after touched
+          // and emit the control
+          this.additionalControls?.forEach((control: NgControl) => {
+            this.subscriptions.push(
+              control?.statusChanges?.subscribe(() => {
```

Comment:
If additional controls get changed, won't we need to unsubscribe from the previous statusChanges subscriptions before we register the new ones?
This may apply for the primary control subscriptions too, if my previous comment gets resolved.

Reply:
Set control is going to happen only once per control if I am not wrong. I doubt that we can change the control i mean replace the existing control with new one. It will happen when container itself is completely destroyed and recreated. Also this is an existing implementation which can be found in other file as well(select-container).

Reply:
Do you think it might become more explicit if we add take(1)?
We won't even need to add it to the subscriptions array if we do.  We're using the take(1) approach in several components and services already.

Reply:
Totally agree, but since its an existing approach in all other files didn't change the approach. Need to do some research why it was added in this way. Hence keeping it in subscriptions mode as of now.

Reply:
My concern is that we're adding more subscriptions inside a subscribe block. This may lead to leaks. It should be okay to add take(1) here. It will make the code safer if it really only gets called once, or will throw explicit error if it unexpectedly gets called more than once, and save us tedious performance/memory-leak debugging.